sdk/identity/azure-identity/inc/azure/identity/azure_cli_credential.hpp

```diff
@@ -19,6 +19,12 @@
 #include <string>
 #include <vector>
 
+#if defined(_azure_TESTING_BUILD)
+namespace Azure { namespace Identity { namespace Test {
+  class AzureCliTestCredential;
```

Comment:
This is not needed - AzureCliTestCredential will be able to inherit this class and overload the virtual function anyway.

Reply:
With the virtual method marked as private, the tests give an accessibility compiler error, for methods like `GetAzCommand`.

> Error	C2248	'Azure::Identity::AzureCliCredential::GetAzCommand': cannot access private member declared in class 'Azure::Identity::AzureCliCredential'
C:\GitHub\Fork\azure-sdk-for-cpp\out\build\x64-static-debug-perftests\azure-sdk-for-cpp
C:\GitHub\Fork\azure-sdk-for-cpp\sdk\identity\azure-identity\test\ut\azure_cli_credential_test.cpp	82